src/pages/index.tsx

```diff
@@ -78,41 +77,35 @@ const Index: FC = () => {
         </div>
     );
 
-    const renderTaskList = () => (
-        <div>
-            {tasksGroupedByStatus[activeTab] ? (
-                <TaskList
-                    tasks={tasksGroupedByStatus[activeTab]}
-                    isEditable={isEditable}
-                    updateCardContent={updateCardContent}
-                />
-            ) : (
-                <p>{NO_TASKS_FOUND_MESSAGE}</p>
-            )}
-        </div>
-    );
-    return (
-        <Layout>
-            <Head title="Tasks" />
-            <TasksProvider>
-                <div className={classNames.container}>
-                    {isError && <p>{TASKS_FETCH_ERROR_MESSAGE}</p>}
-                    {isLoading ? (
-                        <p>Loading...</p>
+    const Content = () => {
```

Comment:
if am not wrong you have declared a react component inside a react component right ?

Reply:
yes but to take it out I have to pass a few props and move the component to another file, the refactoring will go beyond the scope of this PR so I prefer not to do the refactoring in this PR.

Reply:
please make that react component a javascript function it's not good to make a react component inside another react component

Reply:
No, Vaibhav just because it's inside a react component doesn't mean we should make it a function, instead the change required here is we should move it out to a separate file.

Reply:
react components are recreated on each render why do we need to have a react component inside another react component? this is a bad practice in the first place. also as you say the change required is to take it out in a separate file then why wait for the refactor PR ?

Reply:
done